editor/editor_themes.cpp

```diff
@@ -782,6 +783,79 @@ Ref<Theme> create_editor_theme(const Ref<Theme> p_theme) {
 	// FileDialog
 	theme->set_color("files_disabled", "FileDialog", font_color_disabled);
 
+	// adaptive script theme constants
+	// for comments and elements with lower relevance
+	const Color dim_color = Color(font_color.r, font_color.g, font_color.b, 0.5);
+
+	const float mono_value = mono_color.r;
+	const Color alpha1 = Color(mono_value, mono_value, mono_value, 0.1);
+	const Color alpha2 = Color(mono_value, mono_value, mono_value, 0.3);
+	const Color alpha3 = Color(mono_value, mono_value, mono_value, 0.5);
+	const Color alpha4 = Color(mono_value, mono_value, mono_value, 0.7);
+
+	// editor main color
+	const Color main_color = Color::html(dark_theme ? "#57b3ff" : "#0480ff");
+
+	const Color symbol_color = Color::html("#5792ff").linear_interpolate(mono_color, dark_theme ? 0.5 : 0.3);
+	const Color keyword_color = main_color.linear_interpolate(mono_color, 0.4);
+	const Color basetype_color = Color::html(dark_theme ? "#42ffc2" : "#00c161");
+	const Color type_color = basetype_color.linear_interpolate(mono_color, dark_theme ? 0.7 : 0.5);
+	const Color comment_color = dim_color;
+	const Color string_color = Color::html(dark_theme ? "#ffd942" : "#ffd118").linear_interpolate(mono_color, dark_theme ? 0.5 : 0.3);
+
+	const Color te_background_color = Color(0, 0, 0, 0);
+	const Color completion_background_color = base_color;
+	const Color completion_selected_color = alpha1;
+	const Color completion_existing_color = alpha2;
+	const Color completion_scroll_color = alpha1;
+	const Color completion_font_color = font_color;
+	const Color text_color = font_color;
+	const Color line_number_color = dim_color;
+	const Color caret_color = mono_color;
+	// caret_background_color  // what is this for?
```

Comment:
This is the colour of the text behind the caret when `block_caret` is used.

Reply:
good to know